### PR TITLE
Fixed db recommendation chart labels

### PR DESCRIPTION
**Pull request by kema2000**

### CODE REVIEW

**Review by atl-mk — APPROVED** [review]
LGTM. I think it's more consistent but still non-obvious what it is

File: src/test/kotlin/com/atlassian/performance/tools/hardware/report/HardwareExplorationChart.kt [hunk 1]
```diff
@@ -179,7 +180,12 @@ internal class HardwareExplorationChart<S, X>(
                     )
                 ),
                 label = "Recommanded HW : ${recommendation.hardware.jira} x ${recommendation.hardware.nodeCount}",
```
Comment (dagguh) [inline src/test/kotlin/com/atlassian/performance/tools/hardware/report/HardwareExplorationChart.kt]:
```suggestion
                 label = "Recommended HW : ${recommendation.hardware}",
```

File: src/test/kotlin/com/atlassian/performance/tools/hardware/report/HardwareExplorationChart.kt [hunk 2]
```diff
@@ -179,7 +180,12 @@ internal class HardwareExplorationChart<S, X>(
...
-                labelColor = presetLabelColor.color(recommendation.hardware.jira.toString())
+                labelColor = when (seriesGrouping) {
+                    is JiraInstanceTypeGrouping -> presetLabelColor.color(recommendation.hardware.jira.toString())
+                    is JiraClusterGrouping -> presetLabelColor.color(JiraCluster(recommendation.hardware.jira,
+                        recommendation.hardware.nodeCount).toString())
+                    else -> throw Exception("Unknwon grouping : $seriesGrouping")
```
Comment (dagguh) [inline src/test/kotlin/com/atlassian/performance/tools/hardware/report/HardwareExplorationChart.kt]:
```suggestion
                    else -> throw Exception("Unknown grouping : $seriesGrouping")
```

File: src/test/kotlin/com/atlassian/performance/tools/hardware/report/HardwareExplorationChartTest.kt [hunk 3]
```diff
@@ -72,7 +72,7 @@ class HardwareExplorationChartTest {
     @Test
     fun shouldPlotDbRecommendation() {
```
Comment (dagguh) [inline src/test/kotlin/com/atlassian/performance/tools/hardware/report/HardwareExplorationChartTest.kt:73]: The chart title is hardcoded to `Jira XL profile hardware exploration`

Comment (dagguh) [reply src/test/kotlin/com/atlassian/performance/tools/hardware/report/HardwareExplorationChartTest.kt:73]: In general the `hardware-recommendation-chart-template.html` is full of garbage, like chart-js code, Chrome extension code, hardcoded JPT commit etc.
Please clean it up, best to start from scratch: `hardware-exploration-chart-template.html`

**Review by dagguh — CHANGES_REQUESTED** [review]

File: src/test/kotlin/com/atlassian/performance/tools/hardware/report/HardwareExplorationChartTest.kt (hunk 3 above)
Comment (kema2000) [reply src/test/kotlin/com/atlassian/performance/tools/hardware/report/HardwareExplorationChartTest.kt:73]: yep, will do

File: src/test/kotlin/com/atlassian/performance/tools/hardware/report/HardwareExplorationChart.kt (hunk 1 above)
Comment (kema2000) [reply src/test/kotlin/com/atlassian/performance/tools/hardware/report/HardwareExplorationChart.kt]: Thanks, the label is much better now.

**Review by dagguh — APPROVED** [review]

**Comment by dagguh** [comment]
Rebased, squashed

**Pull request merged**